code/game/objects/items/weapons/tools.dm

```diff
@@ -372,75 +345,57 @@
 
 //Returns whether or not the welding tool is currently on.
 /obj/item/weapon/weldingtool/proc/isOn()
-	return src.welding
-
+	return welding
 
 /obj/item/weapon/weldingtool/is_hot()
-	if(isOn())
+	if(welding)
 		return heat_production
 	return 0
 
 
 /obj/item/weapon/weldingtool/is_sharp()
-	if(isOn())
+	if(welding)
 		return sharpness
 	return 0
 
 //Sets the welding state of the welding tool. If you see W.welding = 1 anywhere, please change it to W.setWelding(1)
 //so that the welding tool updates accordingly
-/obj/item/weapon/weldingtool/proc/setWelding(var/temp_welding)
-	//If we're turning it on
-	if(temp_welding > 0)
-		if (remove_fuel(1))
-			to_chat(usr, "<span class='notice'>\The [src] switches on.</span>")
-			src.force = 15
-			src.damtype = "fire"
-			src.icon_state = "welder1"
-			processing_objects.Add(src)
-		else
-			to_chat(usr, "<span class='notice'>Need more fuel!</span>")
-			src.welding = 0
-			return
-	//Otherwise
-	else
-		to_chat(usr, "<span class='notice'>\The [src] switches off.</span>")
-		src.force = 3
-		src.damtype = "brute"
-		src.icon_state = "welder"
-		src.welding = 0
+/obj/item/weapon/weldingtool/proc/setWelding(var/new_welding)
+	if(new_welding != welding)
+		toggle(1)
+	update_icon()
 
 //Turns off the welder if there is no more fuel (does this really need to be its own proc?)
 /obj/item/weapon/weldingtool/proc/check_fuel()
-	if((get_fuel() <= 0) && welding)
-		toggle(1)
+	if(get_fuel() <= 0 && welding)
+		visible_message("<span class='warning'>\The [src] shuts off!</span>")
+		toggle()
+		update_icon()
 		return 0
 	return 1
 
-
 //Toggles the welder off and on
 /obj/item/weapon/weldingtool/proc/toggle(var/message = 0)
 	if(!status)	return
-	src.welding = !( src.welding )
-	if (src.welding)
-		if (remove_fuel(1))
+	welding = !welding
+	if(welding)
+		if(remove_fuel(1))
 			to_chat(usr, "<span class='notice'>You switch the [src] on.</span>")
-			src.force = 15
-			src.damtype = "fire"
-			src.icon_state = "welder1"
+			force = 15
+			sharpness = 0.8
```

Comment:
Why are you modifying sharpness? The point of the get_sharpness proc is that it lets objects do conditional sharpness the way the welder is currently doing, by checking if it's welding for the 0.8 sharpness and then returning 0 otherwise.